static/app/views/alerts/rules/details/metricHistory.tsx

```diff
@@ -0,0 +1,235 @@
+
+  const triggeredActivity = !!criticalActivity.length
+    ? criticalActivity[0]
+    : warningActivity[0];
+  const currentTrigger = getTriggerName(triggeredActivity.value);
+
+  const nextActivity = activities.find(
+    ({previousValue}) => previousValue === triggeredActivity.value
+  );
+
+  const activityDuration = (
+    nextActivity ? moment(nextActivity.dateCreated) : moment()
+  ).diff(moment(triggeredActivity.dateCreated), 'milliseconds');
+
+  const threshold =
+    activityDuration !== null &&
+    tct('[duration]', {
+      duration: <Duration abbreviation seconds={activityDuration / 1000} />,
+    });
+
+  const warningThreshold = incident.alertRule.triggers
+    .filter(trigger => trigger.label === 'warning')
+    .map(trig => trig.alertThreshold);
+  const criticalThreshold = incident.alertRule.triggers
+    .filter(trigger => trigger.label === 'critical')
+    .map(trig => trig.alertThreshold);
+
+  return (
+    <ErrorBoundary>
+      <Title data-test-id="alert-title">
+        <StatusIndicator
+          status={currentTrigger.toLocaleLowerCase()}
+          tooltipTitle={tct('Status: [level]', {level: currentTrigger})}
+        />
+        <Link
+          to={{
+            pathname: alertDetailsLink(organization, incident),
+            query: {alert: incident.identifier},
+          }}
+        >
+          {tct('#[id]', {id: incident.identifier})}
+        </Link>
+      </Title>
+      <Cell>
+        {tct('[title] [selector] [threshold]', {
+          title:
+            AlertWizardAlertNames[getAlertTypeFromAggregateDataset(incident.alertRule)],
+          selector:
+            incident.alertRule.thresholdType === AlertRuleThresholdType.ABOVE
+              ? 'above'
+              : 'below',
+          threshold: currentTrigger === 'Warning' ? warningThreshold : criticalThreshold,
+        })}
+      </Cell>
+      <Cell>{threshold}</Cell>
+      <SeenByCell>
+        <IconWrapper>
+          <Tooltip title={t('People who have viewed this')} skipWrapper>
+            <StyledIconShow size="xs" color="gray200" />
+          </Tooltip>
+          {incident.seenBy.length}
+        </IconWrapper>
+      </SeenByCell>
+      <StyledDateTime date={incident.dateCreated} />
+    </ErrorBoundary>
+  );
+}
+
+type Props = {
+  organization: Organization;
+  incidents?: Incident[];
+};
+
+function MetricHistory({organization, incidents}: Props) {
+  return (
+    <CollapsePanel
+      items={incidents!.length}
+      collapseCount={COLLAPSE_COUNT}
+      disableBorder={false}
+      buttonTitle={tn(
+        'Hidden Alert',
+        'Hidden Alerts',
+        incidents!.length - COLLAPSE_COUNT
+      )}
+    >
+      {({isExpanded, showMoreButton}) => (
+        <div>
+          <StyledPanelTable
+            headers={[
+              t('Alert'),
+              t('Reason'),
+              t('Duration'),
+              t('Seen By'),
+              t('Date Triggered'),
+            ]}
+            isEmpty={!incidents!.length}
```

Comment:
figure out how to avoid doing `incidents!`

Reply:
where you're passing in incidents to these components you might be able to do `incidents ?? []`